pkgs/build-support/fetchmavenartifact/default.nix

```diff
@@ -0,0 +1,57 @@
+# Adaptation of the MIT-licensed work on `sbt2nix` done by Charles O'Farrell
+
+{ fetchurl, stdenv }:
+let
+  defaultRepos = [
```

Comment:
Well, they aren't mirrors, so `mirror://` is not really a good solution. All those urls in `defaultRepos` are some well known Maven repositories, but they don't necessarily contain the same packages. 

I don't think we would gain much by changing the way `repos` is implemented here; I expect people fetching an artifact from a repository other than the ones mentioned in `defaultRepos` will usually have a clear idea where to fetch it from, so they can set `repos` by hand pointing to the specific repository they want.